### PR TITLE
Add possibility to assert a hierarchical graph of spans with parent child relations  

### PR DESCRIPTION
**Pull request by labbati**
### Description

Add the possibility to test a flame graph (with parent child relations) instead of only flat list of spans.

In order to demonstrate how it works, the duplicate elastic search integration test suite has been migrated to the new `assertFlameGraph()` function removing code duplication.

### Readiness checklist
- ~[ ] (only for Members) Changelog has been added to the appropriate release draft. Create one if necessary.~
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the appropriate release draft. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: tests/Integrations/Elasticsearch/V1/ElasticSearchSandboxedIntegrationTest.php [hunk 1]
```diff
@@ -2,270 +2,16 @@
...
-                    // No results, scroll cursor is empty.  You've exported all the data
-                    break;
-                }
-            }
-        });
-
-        $this->assertEmpty($traces);
-    }
 
     public function testScroll()
```
Comment (labbati) [inline tests/Integrations/Elasticsearch/V1/ElasticSearchSandboxedIntegrationTest.php]: An edge case (multiple spans same operation and resource name at the same level) not handled yet. For now we just replicate that single one test and in the future we will improve.

File: tests/Common/SpanChecker.php [hunk 2]
```diff
@@ -2,13 +2,122 @@
 
 namespace DDTrace\Tests\Common;
 
+use DDTrace\Contracts\Span;
```
Comment (SammyK) [inline tests/Common/SpanChecker.php]: This doesn't seem to be referenced anywhere. :)

File: tests/Common/SpanChecker.php [hunk 3]
```diff
@@ -2,13 +2,122 @@
...
+    {
+        $flattenTraces = $this->flattenTraces($traces);
+        $actualGraph = $this->buildSpansGraph($flattenTraces);
+        foreach ($expectedFlameGraph as $oneTrace) {
+            $this->assertNode($actualGraph, $oneTrace);
+        }
+    }
+
+    /**
+     * @param [] $spansByFingerPrint
```
Comment (SammyK) [inline tests/Common/SpanChecker.php]: Just a little nitpick :D 

```suggestion
     * @param array $graph
```

File: tests/Common/SpanChecker.php [hunk 4]
```diff
@@ -2,13 +2,122 @@
...
+     * @return array
+     */
+    private function buildSpansGraph(array $flatSpans)
+    {
+        $byId = [];
+        foreach ($flatSpans as $span) {
+            $byId[$span['span_id']] = ['span' => $span, 'children' => []];
+        }
+
+        while (true) {
```
Comment (SammyK) [inline tests/Common/SpanChecker.php]: Could we change this to a do/while?

```php
do {
// ...
} while (count($byId) !== $lastCount);
```
